plugin-azure-server/src/main/kotlin/jetbrains/buildServer/clouds/azure/arm/AzureCloudClientFactory.kt

```diff
@@ -59,7 +68,9 @@ class AzureCloudClientFactory(cloudRegistrar: CloudRegistrar,
         apiConnector.setServerId(mySettings.serverUUID)
         apiConnector.setProfileId(state.profileId)
 
-        val azureCloudClient = AzureCloudClient(params, apiConnector, myImagesHolder)
+        val myKubePodNameGenerator = KubePodNameGenerator(myEventDispatcher, myExecutorServices)
```

Comment:
Should we pass the generator into the factory instead? _not sure how far up the call tree makes sense_

Reply:
The KubePodNameGenerator class requires the beans EventDispatcher and ExecutorServices. These are injected up the tree into AzureCloudClientFactory (You can check the .xml file in ./resources/META-INF for clarity). It also ensures only one instance of KubePodNameGenerator created per AzureCloudClientFactory instead of multiple KubePodNameGenerator instances per image which is what we want.